src/economy/demographics.cpp

```diff
@@ -97,25 +97,31 @@ void set_employment(sys::state& state, dcon::pop_id p, float v) {
 void set_raw_employment(sys::state& state, dcon::pop_id p, float v) {
 	state.world.pop_set_uemployment(p, to_pu8(v));
 }
-float get_life_needs(sys::state const& state, dcon::pop_id p) {
+template<typename P, typename V>
+V get_life_needs(sys::state const& state, P p) {
```

Comment:
Writing it this way would require you to provide the type of V wherever this function is called since it cannot be deduced from the call site (get_life_needs<float>(...) for example). Instead of V, you can write `auto` for the return type which will cause the compiler to deduce the return type from the function body